File: tests/fixtures/step-defs/go/auth_steps.go
package steps

import (
	"github.com/cucumber/godog"
)

func InitializeScenario(s *godog.ScenarioContext) {
	s.Given(`a registered user`, aRegisteredUser)
	s.When(`the user logs in`, theUserLogsIn)
	s.Then(`the user should see the dashboard`, shouldSeeDashboard)
}

func aRegisteredUser(ctx context.Context) error {
	db.CreateUser("testuser", "secret123")
	return nil
}

func theUserLogsIn(ctx context.Context) error {
	response = app.Login("testuser", "secret123")
	return nil
}

func shouldSeeDashboard(ctx context.Context) error {
	if response.Status != 200 {
		return fmt.Errorf("expected 200, got %d", response.Status)
	}
	return nil
}

File: tests/fixtures/step-defs/java/AuthSteps.java
package com.example.steps;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.When;
import io.cucumber.java.en.Then;
import static org.junit.Assert.*;

public class AuthSteps {

    @Given("a registered user")
    public void aRegisteredUser() {
        User user = new User("testuser", "secret123");
        db.save(user);
    }

    @When("the user logs in with valid credentials")
    public void theUserLogsIn() {
        response = app.login("testuser", "secret123");
    }

    @Then("the user should see the dashboard")
    public void shouldSeeDashboard() {
        assertEquals(200, response.getStatus());
        assertTrue(response.getBody().contains("dashboard"));
    }
}
